spiped/main.c

```diff
@@ -357,7 +361,8 @@ main(int argc, char * argv[])
 err4:
 	free(K);
 err3:
-	sock_addr_freelist(sas_t);
+	if (sas_t != NULL)
```

Comment:
I just noticed that `sock_addr_freelist()` does nothing if the pointer is already NULL, so this line isn't strictly necessary.